js/src/dapps/registry/ui/address.js

```diff
@@ -14,34 +14,67 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 
-import React from 'react';
-import renderHash from './hash';
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+import Hash from './hash';
+import etherscanUrl from '../util/etherscan-url';
 import IdentityIcon from '../IdentityIcon';
 
-const container = {
-  display: 'inline-block',
-  verticalAlign: 'middle',
-  height: '24px'
-};
-const align = {
-  display: 'inline-block',
-  verticalAlign: 'top',
-  lineHeight: '24px'
-};
+import styles from './address.css';
 
-export default (address, accounts, contacts, shortenHash = true) => {
+const Address = ({ address, accounts, contacts, isTestnet, key, shortenHash }) => {
   let caption;
-  if (accounts[address]) {
-    caption = (<abbr title={ address } style={ align }>{ accounts[address].name || address }</abbr>);
-  } else if (contacts[address]) {
-    caption = (<abbr title={ address } style={ align }>{ contacts[address].name || address }</abbr>);
+  if (accounts[address] || contacts[address]) {
+    const name = (accounts[address] || contacts[address] || {}).name;
+    caption = (
+      <a
+        className={ styles.link }
+        href={ etherscanUrl(address, isTestnet) }
+        target='_blank'
+      >
+        <abbr title={ address } className={ styles.align }>
+          { name || address }
+        </abbr>
+      </a>
+    );
   } else {
-    caption = (<code style={ align }>{ shortenHash ? renderHash(address) : address }</code>);
+    caption = (
+      <code className={ styles.align }>
+        { shortenHash ? (<Hash hash={ address } linked />) : address }
+      </code>
+    );
   }
+
   return (
-    <div style={ container }>
-      <IdentityIcon address={ address } style={ align } />
+    <div key={ key } className={ styles.container }>
+      <IdentityIcon address={ address } className={ styles.align } />
       { caption }
     </div>
   );
 };
+
+Address.propTypes = {
```

Comment:
I guess this is not the place to discuss functional vs `class` components, so I will adapt it. 😉